src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+  labs(title = "No normalization", x = '', color = "Environment") +
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 6))
+
+# Plot with rarefied data
+alpha_diversity_rarefy <- ggplot(rarefied_melted_df, aes(x = Sample, y = value, color = Sample)) +
+  geom_boxplot() +
+  facet_wrap(~ measure, scales = "free_y") +
+  theme_bw() +
+  scale_color_manual(values = viridis(40)) +
+  labs(title = "Rarefy", x = '', color = "Environment") +
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 6))
+
+# Plot with css normalization
+alpha_diversity_css <- ggplot(css_melted_df, aes(x = Sample, y = value, color = Sample)) +
+  geom_boxplot() +
+  facet_wrap(~ measure, scales = "free_y") +
+  theme_bw() +
+  scale_color_manual(values = viridis(40)) +
+  labs(title = "CSS Normalization", x = '', color = "Environment") +
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size = 6))
+```
+
+## Community profile visualization
+Here we visualise the taxonomic profiles in barplots at class ranks for our top tss normalized taxa, rarefied taxa and css normalized taxa.
+
+1. Lets change any unknown taxonomic rowDate information to "Unknown" from "NA".
+
+```{r}
+# Changing rowData on tss_tse object
+rowData(tss_tse)$Class <- ifelse(is.na(rowData(tss_tse)$Class), "Unknown", rowData(tss_tse)$Class)
+
+# Changing rowData on rarefy_tse object
+rowData(rarefy_tse)$Class <- ifelse(is.na(rowData(rarefy_tse)$Class), "Unknown", rowData(rarefy_tse)$Class)
+
+# Changing rowData on css_tse object
+rowData(css_tse)$Class <- ifelse(is.na(rowData(css_tse)$Class), "Unknown", rowData(css_tse)$Class)
+```
+
+2. Create the plots with the `plotAbundance` `mia` method.
+
+```{r}
+# Plotting our tss data
+tss_plot <- plotAbundance(tss_tse,
+                            assay.type = "tss",
+                            rank = "Class",
+                            order_rank_by = "abund")
+
+# Plotting our rarefied data
+rarefy_plot <- plotAbundance(rarefy_tse,
+                               assay.type = "rarefy",
+                               rank = "Class",
+                               order_rank_by = "abund")
+
+# Plotting our css data
+css_plot <- plotAbundance(css_tse,
+                             assay.type = "css",
+                             rank = "Class",
+                             order_rank_by = "abund")
+```
+
+By naked eye, we can see that the profiles change. Let’s take a deeper look at some interesting Classes to check how the abundance change depending on the normalization method.
+
+1. Define our subset classes
+
+```{r}
+subset_classes <- c("Betaproteobacteria", 
+                    "Bacteroidia", 
+                    "Actinobacteria")
+```
+
+
+2. Subset our TreeSummarizedExperiment objects.
+
+```{r}
+# Function to subset TreeSummarizedExperiment object
+subset_tse <- function(tse, classes) {
+  subset_indices <- which(rowData(tse)$Class %in% classes)
+  tse[subset_indices, ]
+}
```

Comment:
Do we need separate function for this? Why not just do `tse[rowData(tse)$Class %in classes, ]` - it is not any longer